components/engine_rocks/src/util.rs

```diff
@@ -331,6 +333,137 @@ pub fn from_raw_perf_level(level: rocksdb::PerfLevel) -> engine_traits::PerfLeve
     }
 }
 
+struct KeyRange {
+    start_key: Box<[u8]>,
+    end_key: Box<[u8]>,
+}
+
+pub struct NoopFilter;
+
+impl CompactionFilter for NoopFilter {}
+
+pub struct NoopFactory;
+
+impl CompactionFilterFactory for NoopFactory {
+    type Filter = NoopFilter;
+    fn create_compaction_filter(
+        &self,
+        _context: &CompactionFilterContext,
+    ) -> Option<(CString, Self::Filter)> {
+        None
+    }
+
+    fn should_filter_table_file_creation(&self, reason: DBTableFileCreationReason) -> bool {
+        matches!(
+            reason,
+            DBTableFileCreationReason::Flush | DBTableFileCreationReason::Compaction
+        )
+    }
+}
+
+/// Used to build `RangeCompactionFilterFactory`, optionally stacked over
+/// another compaction filter factory.
+pub struct RangeCompactionFilterFactoryBuilder(Arc<KeyRange>);
+
+impl RangeCompactionFilterFactoryBuilder {
+    pub fn new(start_key: Vec<u8>, end_key: Vec<u8>) -> Self {
```

Comment:
Why not accept `Box<[u8]>` directly?

Reply:
Because it is rare that caller will have a boxed slice instead of a vec. To accept a box will add some boilerplate code to all caller sites.

Reply:
I can see the only place calling the function is in engine factory where key is already defined as `Box<[u8]>`. If it's `Vec<u8>`, then calling `.into()` is enough.